lrauv_ignition_plugins/src/VisualizePointCloud.cc

```diff
@@ -51,32 +54,26 @@ namespace tethys
     /// \brief Transport node
     public: ignition::transport::Node node;
 
-    /// \brief Science data type-specific topic name to subscribe
-    public: std::string topicName{""};
+    /// \brief Name of topic for PointCloudPacked
+    public: std::string pointCloudTopic{""};
 
-    /// \brief List of science data topics
-    public: QStringList topicList;
+    /// \brief Name of topic for FloatV
+    public: std::string floatVTopic{""};
 
-    /// \brief Protect variables changed from transport and the user
-    public: std::recursive_mutex mutex;
+    /// \brief List of topics publishing PointCloudPacked.
+    public: QStringList pointCloudTopicList;
 
-    /// \brief Generic point cloud topic name
-    public: std::string pcTopic = {"/science_data"};
+    /// \brief List of topics publishing FloatV.
+    public: QStringList floatVTopicList;
 
-    /// \brief Generic point cloud service name
-    public: std::string pcSrv = {"/science_data_srv"};
+    /// \brief Protect variables changed from transport and the user
+    public: std::recursive_mutex mutex;
 
     /// \brief Point cloud message containing location of data
-    public: ignition::msgs::PointCloudPacked pcMsg;
-
-    /// \brief Temperature data to visualize
-    public: ignition::msgs::Float_V tempMsg;
+    public: ignition::msgs::PointCloudPacked pointCloudMsg;
 
-    /// \brief Chlorophyll data to visualize
-    public: ignition::msgs::Float_V chlorMsg;
-
-    /// \brief Salinity data to visualize
-    public: ignition::msgs::Float_V salMsg;
+    /// \brief Message holding a float vector.
+    public: ignition::msgs::Float_V floatVMsg;
```

Comment:
I think we should let the user have the capability of visualizing more than one type of data at a time, because sometimes it's useful to overlap multiple modalities and see how they relate. For example, temperature and salinity together directly imply density, while either one of them alone doesn't.

We might add the capability of adding more topic drop-down lists, rather than restricting to subscribing only to one topic at a time.

Reply:
If we don't want to hardcode them, they can be in a vector that expands and contracts. I can open an issue about that.

Reply:
Speaking of not hardcoding, I saw there was negative data in chlorophyll and wasn't sure if it's an error or intended. So I capped off at 0 and visualized everything below 0 at the same color as 0. Looks like in this subset of data we're visualizing, we don't have negatives. Just curious @braanan are those values intended to be negative?

Reply:
> we should let the user have the capability of visualizing more than one type of data at a time

Oh sorry, I was going to add a gif but forgot. That use case is already supported, here it goes:

![multiple_data](https://user-images.githubusercontent.com/5751272/142669871-a14b9f36-5866-4db9-84f1-ea39f5604f45.gif)

Reply:
Awesome!